### ISSUE TITLE
[BUG] Compute mistakenness cannot handle None-valued predictions

### ISSUE THREAD

**Issue by Rusteam** [voxel51/fiftyone#1543]
### Proposal Summary

`fob.compute_mistakenness()` raises an error if it encounters a sample with `None`-valued predictions.

### Motivation

-   What is the use case for this feature?

My steps are:
1. I load my yolov5 dataset
2. and then load model predictions as a new field `prediction`
3. and run `dataset.evaluate_detections` and `fob.compute_mistakenness`
4. but it fails because some samples from the original dataset have None at prediction field
`ValueError: Sample '61e663228baaa140f1ab0a67' field 'prediction' is None`
since there're not corresponding detections file

-   Why is this use case valuable to support for FiftyOne users in general?

If we skip these examples we won't know that there're false negatives in detections.
Then this leads to erroneous metric assessment and is somewhat hidden.

-   Why is this use case valuable to support for your project(s) or
    organization?

no extra steps required when comparing ground truth and predictions like manually correcting the `prediction` field.

-   Why is it currently difficult to achieve this use case? (please be as
    specific as possible about why related FiftyOne features and components are
    insufficient)

Not difficult at all. Create a view with prediction is None and assign empty detections.

### What areas of FiftyOne does this feature affect?

-   [ ] App: FiftyOne application
-   [x] Core: Core `fiftyone` Python library
-   [ ] Server: FiftyOne server

### Details

yolo.py

```
def add_yolo_labels(sample_collection, label_field, labels_path, classes=None,
                        include_missing=False # new param):
...

view = sample_collection.select(matched_ids, ordered=True)
labels = [load_yolo_annotations(p, classes) for p in matched_paths]
view.set_values(label_field, labels)

# if True then set empty
if include_missing:
    missing_view = sample_collection.exclude(matched_ids, ordered=True)
    missing_labels = [fo.Detections(detections=[]) ] * len(missing_view)
    missing_view.set_values(label_field, missing_labels) 
```

### Willingness to contribute

The FiftyOne Community encourages new feature contributions. Would you or
another member of your organization be willing to contribute an implementation
of this feature?

-   [x] Yes. I can contribute this feature independently.
-   [ ] Yes. I would be willing to contribute this feature with guidance from
        the FiftyOne community.
-   [ ] No. I cannot contribute this feature at this time.

**Comment by brimoor** [comment]
Hey @Rusteam thanks for the suggestion.

First, we should definitely update `compute_mistakenness()` to treat `None` as empty (I believe `evaluate_detections()` already does this), to capture false negatives as you mention. If you include such a sample in your method call, you are definitely intending to treat `None` <-> `Detections()`. 

In general, I'm hesitant to add empty `Detections()` in `add_yolo_labels()` because in some use cases it is meaningful to distinguish:
- `Detections()`: the sample was included in annotation/prediction, but no labels were detected
-  `None`: the sample was not processed, so it may or may not contain objects. These samples should generally be omitted from any analysis via `dataset.exists(field)`

The recommended way to cause `add_yolo_labels()` to insert `Detections()` instead of `None` is to have empty-but-existing TXT files in your labels directory.

As a workaround, you can insert empty `Detections()` in your dataset like so:
 
```py
missing_preds = dataset.exists("predictions", False)
missing_preds.set_values("predictions", [fo.Detections()] * len(missing_preds))
```

**Comment by brimoor** [comment]
Update: oh I see now that in your proposal the default `include_missing=False` maintains the current behavior.

I'm onboard with having your idea as an optional behavior at sample add-time. A PR is welcome! 🤗 Could you also add the same thing to `fiftyone.utils.add_coco_labels()`?

**Comment by Rusteam** [comment]
> Update: oh I see now that in your proposal the default `include_missing=False` maintains the current behavior.
> 
> I'm onboard with having your idea as an optional behavior in `add_yolo_labels()`. A PR is welcome! 🤗 Could you also add the same thing to `fiftyone.utils.add_coco_labels()`?

Yeah I can do that.

Also let me figure out how to handle None's in `compute_mistakeness`

**Comment by brimoor** [comment]
Awesome! FYI `compute_mistakeness()` is from a separate `fiftyone-brain` package, so we'll need to handle that separately.

**Comment by Rusteam** [comment]
> the same thing to fiftyone.utils.add_coco_labels()?

seems like it is a default behavior for coco

**Comment by brimoor** [comment]
`compute_mistakeness()` has been updated to gracefully handle None values, and this behavior will be available in `fiftyone-brain>0.7.3`.

**Comment by Rusteam** [comment]
thanks a lot
